Add Phabricator e2e test script

lguychard opened the pull request:
Closes #3667

felixfbecker commented:
@chrismwendt you're owning this for 3.5, right?

chrismwendt commented:
@felixfbecker Yes, I plan to get back to this after https://github.com/sourcegraph/sourcegraph/issues/4051

felixfbecker commented:
👍

lguychard commented on web/src/e2e/util.ts:
```diff
@@ -238,9 +322,15 @@ export async function createDriverForTest(): Promise<Driver> {
+    const launchOpt: LaunchOptions & { appMode: boolean } = {
         args: [...args, '--window-size=1280,1024'],
         headless: readEnvBoolean({ variable: 'HEADLESS', defaultValue: false }),
+        defaultViewport: null,
```
This is needed so that puppeteer correctly displays a 1280*1024 viewport. In the default 800*600 viewport, some buttons on Phabricator are not shown, which breaks the e2e test.

lguychard commented on web/src/e2e/util.ts:
```diff
@@ -143,15 +151,13 @@ export class Driver {
         await this.replaceText({
             selector: '#e2e-external-service-form-display-name',
             newText: displayName,
-            enterTextMethod: 'paste',
```
`ensureHasExternalService()` fails locally every time for me unless I delete this line 🤔

lguychard commented:
@chrismwendt @felixfbecker I've rebased/fixed this so that we can merge it & use it during the next release testing cycle instead of testing Phabricator manually.

- I've moved the e2e test to `web/`, to use the same `Driver` apis as the e2e & regression test suites now use. Since the test suite is run against a Sourcegraph instance, and not a browser extension, I think this makes sense.
- I moved `ensureHasCORSOrigin()` & its dependencies to `Driver`, instead of having them defined in a separate utils file like they were originally, for consistency (since it's similar to `ensureLoggedIn()`, `ensureHasExternalService`...).
- I haven't set this up to run in CI since it'd add 5+ minutes to build times. This can be run using the new `./dev/phabricator/e2e.sh` script. I've successfully tested it against the Sourcegraph 3.6.0 docker image: `SOURCEGRAPH_BASE_URL=http://localhost:7080 ./dev/phabricator/e2e.sh`.

beyang commented on dev/phabricator/e2e.sh:
```diff
@@ -0,0 +1,12 @@
+set -e
```
@lguychard can you add yourself as owner of the new files in `.github/CODEOWNERS`?

felixfbecker commented:
@chrismwendt, just to check, you are owning to bring this over the finish line for 3.7?

chrismwendt commented:
I squashed and rebased on the latest master and fixed a couple bugs in the script. The test passes now.

chrismwendt commented on web/src/e2e/phabricator.e2e.test.tsx:
```diff
@@ -0,0 +1,139 @@
+    await addPhabricatorRepo(driver)
+}
+
+describe('Sourcegraph Phabricator extension', () => {
```
The location could be argued both ways. If anyone feels strongly one way or the other, feel free to move it.
